test/routes/sessions-google.js

```diff
@@ -10,64 +11,92 @@ chai.use(chaiSubset);
 
 const envSecret = Array(65).join('0');
 const authSecret = Array(65).join('1');
+const googleAccessToken = 'google-access-token';
 
 describe('API > Google OAuth2 Login', () => {
   let app;
-  let googleServiceData;
+  let sandbox;
 
   before(() => {
-    const dependencies = {
-      GoogleAuthorizationFinder: function (renderingId, forestToken, envSecret) {
-        googleServiceData = { renderingId, forestToken, envSecret };
+    sandbox = sinon.createSandbox();
+    const forestServerRequester = require('../../src/services/forest-server-requester');
 
-        this.perform = function perform() {
-          return P.resolve({
-            id: 'id',
+    app = createServer(envSecret, authSecret);
+
+    const stubPerform = sandbox.stub(forestServerRequester, 'perform');
+
+    stubPerform.withArgs('/liana/v1/ip-whitelist-rules').returns({
+      then: () => P.resolve({
+        data: {
+          attributes: {
+            use_ip_whitelist: false,
+          },
+        },
+      }),
+    });
+
+    stubPerform.withArgs(
+      '/liana/v2/renderings/1/google-authorization',
+      envSecret,
+      null,
+      { 'forest-token': googleAccessToken },
+    ).resolves({
+      status: 200,
+      body: {
+        data: {
+          id: '654',
+          type: 'users',
+          attributes: {
             email: 'user@email.com',
             first_name: 'FirstName',
             last_name: 'LastName',
-            teams: [],
-          });
-        };
+            teams: ['Operations'],
+          },
+        },
+        relationships: {
+          renderings: {
+            data: [{
+              id: 1,
+              type: 'renderings',
+            }],
+          }
+        },
       },
+    });
+  });
 
-      ApimapSender: function () {
-        this.perform = function () {};
-      }
-    };
-
-    app = createServer(envSecret, authSecret, dependencies);
+  after(() => {
+    sandbox.restore();
   });
 
   describe('POST /forest/sessions-google', () => {
     it('should return a valid jwt token', (done) => {
       request(app)
         .post('/forest/sessions-google')
-        .send({ renderingId: 1, forestToken: 'google-access-token' })
+        .send({ renderingId: 1, forestToken: googleAccessToken })
         .expect(200)
         .end((error, response) => {
           expect(error).to.be.null;
 
-          expect(googleServiceData).to.containSubset({
-            renderingId: 1,
-            forestToken: 'google-access-token',
-            envSecret,
-          });
-
           const { token } = response.body;
           const decodedJWT = jsonwebtoken.verify(token, authSecret);
 
           expect(decodedJWT).to.containSubset({
```

Comment:
The test fails:
<img width="934" alt="capture d ecran 2018-09-21 a 14 32 19" src="https://user-images.githubusercontent.com/604486/45881390-4e914280-bdab-11e8-94ab-793eb041f53e.png">

Reply:
Yes, sorry, this is the bug we currently have and a supposed to fix in the next PR to be merged.